packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -160,14 +169,17 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     @required AxisDirection crossAxisDirection,
     @required ViewportOffset offset,
     double cacheExtent,
+    CacheExtentStyle cacheExtentStyle = CacheExtentStyle.pixel,
   }) : assert(axisDirection != null),
        assert(crossAxisDirection != null),
        assert(offset != null),
        assert(axisDirectionToAxis(axisDirection) != axisDirectionToAxis(crossAxisDirection)),
+       assert(cacheExtentStyle != null),
        _axisDirection = axisDirection,
        _crossAxisDirection = crossAxisDirection,
        _offset = offset,
-       _cacheExtent = cacheExtent ?? RenderAbstractViewport.defaultCacheExtent;
+       _cacheExtent = cacheExtent ?? RenderAbstractViewport.defaultCacheExtent,
```

Comment:
The default should probably depend on the _cacheExtentStyle? E.g. if you don't set an explicit _cacheExtend, but set the style to viewport, we don't want to cache 250 pages.

Reply:
I'm not allowing _cacheExtent to be null if _cacheExtentStyle is viewport.  I'm not even sure what a default should be in that case, - .25? 1.0?